GetProfileActivity/handler.ts

```diff
@@ -113,46 +115,48 @@ export const createGetProfileActivityHandler = (profileModel: ProfileModel) => (
   context: Context,
   input: unknown
 ) =>
-  // the actual handler
-  fromEither(ActivityInput.decode(input))
-    .mapLeft<ActivityResultFailure>((reason: t.Errors) =>
+  pipe(
+    input,
+    ActivityInput.decode,
+    // the actual handler
+    E.mapLeft(reason =>
       ActivityResultInvalidInputFailure.encode({
         kind: "INVALID_INPUT_FAILURE",
         reason: readableReport(reason)
       })
-    )
-    .chain(({ fiscalCode }) =>
-      profileModel
-        .findLastVersionByModelId([fiscalCode])
-        .foldTaskEither<ActivityResultFailure, RetrievedProfile>(
-          error =>
-            fromLeft(
-              ActivityResultQueryFailure.encode({
-                kind: "QUERY_FAILURE",
-                query: "findLastVersionByModelId",
-                reason: `${error.kind}, ${getMessageFromCosmosErrors(error)}`
+    ),
+    TE.fromEither,
+    TE.chainW(({ fiscalCode }) =>
+      pipe(
+        profileModel.findLastVersionByModelId([fiscalCode]),
+        TE.mapLeft(error =>
+          ActivityResultQueryFailure.encode({
+            kind: "QUERY_FAILURE",
+            query: "findLastVersionByModelId",
+            reason: `${error.kind}, ${getMessageFromCosmosErrors(error)}`
+          })
+        ),
+        TE.chainW(
+          flow(
```

Comment:
`TE.fromOption` should suffice here?

Reply:
fixed in https://github.com/pagopa/io-functions-admin/pull/165/commits/f78dd8d8855669542c1b23ad9168b014f4a4000a